providers/vsphere/provider/provider.go

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package provider
+
+import (
+	"errors"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"go.mondoo.com/cnquery/providers-sdk/v1/vault"
+
+	"go.mondoo.com/cnquery/llx"
+	"go.mondoo.com/cnquery/providers-sdk/v1/inventory"
+	"go.mondoo.com/cnquery/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/providers-sdk/v1/upstream"
+	"go.mondoo.com/cnquery/providers/vsphere/connection"
+	"go.mondoo.com/cnquery/providers/vsphere/resources"
+)
+
+type Service struct {
+	runtimes         map[uint32]*plugin.Runtime
+	lastConnectionID uint32
+}
+
+func Init() *Service {
+	return &Service{
+		runtimes:         map[uint32]*plugin.Runtime{},
+		lastConnectionID: 0,
+	}
+}
+
+func (s *Service) ParseCLI(req *plugin.ParseCLIReq) (*plugin.ParseCLIRes, error) {
+	flags := req.Flags
+	if flags == nil {
+		flags = map[string]*llx.Primitive{}
+	}
+
+	conn := &inventory.Config{
+		Type: req.Connector,
+	}
+
+	// parse args
+	user := ""
+	port := 0
```

Comment:
Has vsphere a default port we could use?

Reply:
definitely a good follow-up! i know this code has mostly just migrated what was already there

Reply:
We do not care in cli parsing, defaults should happen in the connection itself. Otherwise missing values in inventories would not work. The handling for 0 ports is here https://github.com/mondoohq/cnquery/pull/1551/files#diff-719e7f1b1168518ce8f65b0279ac8c2892e64752122488e50d36241a7c0199abR28-R30